src/controllers/auth.controller.js

```diff
@@ -0,0 +1,129 @@
+const { userSchema } = require('../models/user');
+
+const { authServices } = require('../services/auth.services');
+const { jwtServices } = require('../services/jwt.services');
+const { tokenServices } = require('../services/token.services');
+const { usersServices } = require('../services/users.services');
+const {
+  sendActivationLink,
+  sendResetLink,
+} = require('../services/email.services');
+
+const { ApiError } = require('../exeptions/auth.error');
+
+const { checkPassword } = require('../utils/password/checkPassword');
+const { asyncHandler } = require('../utils/asyncHandler');
+
+const authValidation = async (req, _, next) => {
+  const { email, password } = req.body;
+
+  try {
+    await userSchema.validate({ email, password });
+  } catch (e) {
+    throw ApiError.badRequest(e.message);
```

Comment:
The `ApiError` constructor arguments are incorrectly ordered. Ensure the correct order is `message, status, errors`.